### PR TITLE
imx: fix race condition in interrupt handling 

### PR DESCRIPTION
**Pull request by iuliana-prodan**
See each commit message

### CODE REVIEW

**Review by paulstelian97 — APPROVED** [review]

**Review by dbaluta — CHANGES_REQUESTED** [review]
Let's wait for more testing. I have a different patch for this.

**Comment by iuliana-prodan** [comment]
> Let's wait for more testing. I have a different patch for this.

@dbaluta this was tested on 2 different 8QM boards with both SRC and MIXER tplgs, with suspend-resume.
The tests were run 5 times in a row, with no failure or crash.

**Comment by iuliana-prodan** [comment]
I've added a 3rd commit to this PR to fix qemu boot failure on imx8 (this was due to commit 1 in which I've updated the interrupt initialization sequence).

File: src/drivers/imx/ipc.c [hunk 1]
```diff
@@ -69,7 +69,6 @@ static void irq_handler(void *arg)
 
 	/* new message from host */
 	if (status & IMX_MU_xSR_GIPn(IMX_MU_VERSION, 0)) {
-
 		/* Disable GP interrupt #0 */
```
Comment (dbaluta) [inline src/drivers/imx/ipc.c:73]: this seems unrelated. Is this intentional?

**Review by dbaluta — APPROVED** [review]

**Comment by dbaluta** [comment]
Overnight tests passed. Excellent work Iulia!

**Comment by iuliana-prodan** [comment]
> Overnight tests passed. Excellent work Iulia!

Thanks! It was a joint work!

**Pull request merged**